doc/en/user/source/security/csp.rst

```diff
@@ -0,0 +1,270 @@
+the ``frame-ancestors`` directive. This is intended to make it easier for administrators to allow
+specific remote hosts to load GeoServer content in frames. Only trusted hosts should be added
+here to prevent clickjacking attacks. The ``geoserver.csp.frameAncestors`` system property will
+override this field if it has been set. Setting this to ``'self' https://geoserver.org`` would
+change the header value at the top of this page to::
+
+    base-uri 'self'; form-action 'self'; default-src 'none'; child-src 'self'; connect-src 'self'; font-src 'self'; img-src 'self' data:; style-src 'self' 'unsafe-inline'; script-src 'self';, frame-ancestors 'self' https://geoserver.org;
+
+Configuring Policies
+--------------------
+
+Each policy contains the rules and directives to set a single CSP header value. When there are CSP
+directives from multiple policies, the directives will be concatenated into a single line using
+commas rather than setting multiple Content-Security-Policy headers. If a value for a specific
+directive is defined in multiple policies, web browsers will use the strictest value set for that
+directive.
+
+The button for adding policies can be found at the top of the :guilabel:`Policy List` table and a
+policy can be edited by clicking on its name in the table or removed by clicking on the remove icon
+at the end of the policy's row in the table. Policy positions can also be changed by using the
+up/down arrows or by dragging and dropping the rule's row in the table.
+
+.. figure:: images/csp-policy.png
+
+   CSP Policy Configuration
+
+* A unique name must be provided in the :guilabel:`Name` text field when adding a new policy.
+* The :guilabel:`Description` text field provides an optional description to help administrators
+  understand what the policy does.
+* The :guilabel:`Enabled` checkbox will enable/disable the policy.
+
+.. note::
+    After saving a policy, make sure to save/apply the configuration.
+
+Configuring Rules
+-----------------
+
+Each rule contains a filter to match against user requests and the CSP directives to add to the
+header value for matching requests. Rules will be checked in order against incoming requests and
+only the first matching rule in each policy will be applied. If no rule in a policy matches the
+request, then no directives will be added to the CSP header from that policy. If the matching
+rule has no directives defined, then preceding rules will be checked until the first rule is
+found that has directives and no directives will be added if no such rule exists.
+
+The button for adding rules can be found at the top of the :guilabel:`Rule List` table and a rule
+can be edited by clicking on its name in the table or removed by clicking on the remove icon at the
+end of the rule's row in the table. Rule positions can also be changed by using the up/down arrows
+or by dragging and dropping the rule's row in the table.
+
+.. figure:: images/csp-rule.png
+
+   CSP Rule Configuration
```

Comment:
We need a new example as the WFS Test Servlet is not included any more.

Reply:
Done